Source/Core/DolphinQt/TAS/BalanceBoardWidget.cpp

```diff
@@ -0,0 +1,93 @@
+// Copyright 2018 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "DolphinQt/TAS/BalanceBoardWidget.h"
+
+#include <algorithm>
+#include <cmath>
+
+#include <QMouseEvent>
+#include <QPainter>
+
+#include "Common/CommonTypes.h"
+
+BalanceBoardWidget::BalanceBoardWidget(QWidget* parent) : QWidget(parent)
+{
+  setMouseTracking(false);
+  setToolTip(tr("Left click to set the IR value.\n"
+                "Right click to re-center it."));
+}
+
+void BalanceBoardWidget::SetX(u16 x)
+{
+  m_x = std::min(balance_range, x);
+
+  update();
+}
+
+void BalanceBoardWidget::SetY(u16 y)
+{
+  m_y = std::min(balance_range, y);
+
+  update();
+}
+
+void BalanceBoardWidget::paintEvent(QPaintEvent* event)
+{
+  QPainter painter(this);
+
+  painter.setRenderHint(QPainter::Antialiasing, true);
+  painter.setRenderHint(QPainter::SmoothPixmapTransform, true);
+
+  painter.setBrush(Qt::white);
+  painter.drawRect(0, 0, width() - 1, height() - 1);
+
+  painter.drawLine(0, height() / 2, width(), height() / 2);
+  painter.drawLine(width() / 2, 0, width() / 2, height());
+
+  // convert from value space to widget space
+  u16 x = (m_x * width()) / balance_range;
+  u16 y = height() - (m_y * height()) / balance_range;
+
+  painter.drawLine(width() / 2, height() / 2, x, y);
+
+  painter.setBrush(Qt::blue);
+  int wh_avg = (width() + height()) / 2;
+  int radius = wh_avg / 30;
+  painter.drawEllipse(x - radius, y - radius, radius * 2, radius * 2);
```

Comment:
These calculations are intermixing signed and unsigned types. Please don't do this. Even if it results in the correct values, it can cause compilation warnings.